Purchases/SwiftSources/LocalReceiptParsing/BasicTypes/ASN1Container.swift

```diff
@@ -0,0 +1,63 @@
+//
+// Created by Andrés Boedo on 7/28/20.
+// Copyright (c) 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+enum ASN1Class: UInt8 {
+    case universal, application, contextSpecific, `private`
+}
+
+enum ASN1Type: UInt8 {
+    case endOfContent = 0x00
+    case boolean = 1
+    case integer = 2
+    case bitString = 3
+    case octetString = 4
+    case null = 5
+    case objectIdentifier = 6
+    case objectDescriptor = 7
+    case external = 8
+    case real = 9
+    case enumerated = 10
+    case embeddedPdv = 11
+    case utf8String = 12
+    case relativeOid = 13
+    case sequence = 16
+    case set = 17
+    case numericString = 18
+    case printableString = 19
+    case t61String = 20
+    case videotexString = 21
+    case ia5String = 22
+    case utcTime = 23
+    case generalizedTime = 24
+    case graphicString = 25
+    case visibleString = 26
+    case generalString = 27
+    case universalString = 28
+    case characterString = 29
+    case bmpString = 30
+}
+
+enum ASN1EncodingType: UInt8 {
+    case primitive, constructed
+}
+
+struct ASN1Length {
+    let value: UInt
+    let totalBytes: Int
```

Comment:
no good reason tbh. At first I was using UInt a lot more since the inability for the values to be negative made sense, but it's a total hassle because most useful methods require `Int` (like the array slicing methods `dropFirst`, `prefix`, etc). 
I'll refactor `value` to `Int` since it removes the need for quite a bit of casting.